modules/installation/pages/upgrade.adoc

```diff
@@ -85,6 +85,128 @@ $ gadmin restart all -y
+
+[NOTE]
+====
+Rolling Upgrade is only available for HA clusters (High Availability) and is supported for maintenance releases (e.g., 4.2.0 to 4.2.1, 4.3.1 to 4.3.3). 
+Upgrading from 4.2.0 to 4.3.0 or any other major version is not supported.
+====
+
+During the rolling upgrade, certain operations are not supported, including installing queries, running schema changes, dropping all data, using `gsql --reset`, creating graphs, backup/restore, clearing graph stores, exporting/importing graphs, and installing functions. Additionally, occasional drops in queries per second (QPS) may occur. 
+The following services will not be upgraded during the rolling upgrade process: Zookeeper (ZK), Kafka, and ETCD.
+
+=== Rolling Upgrade Process
+
+. Download and Extract Latest Version
+
+* Download the latest maintenance release of TigerGraph.
+* Extract the tar file on your system.
+
+. Run Installation Script with Upgrade Flag
+
+Run the following installation script with the upgrade flag (`+-Ur+`) that was extracted from the tar file:
+
+[source, gsql]
+----
+$ ./install.sh -Ur
+----
+
+. Switching to New Version
+
+After the binaries and configuration files are installed, user will be prompted with the following message:
+
+[console]
+----
+"Do you want to switch platform to the new version now (it can be delayed to a later time): y/N"
+----
+
+. n/N option
+
+Selecting n/N will guide the user through the manual steps to switch to version 4.2.1, assuming the user is upgrading from 4.2.0.
+
+[console]
+----
+To switch platform to new version, please run 'gadmin upgrade 4.2.1'
+----
+
+. y/Y Option
+
+If users selects `y/Y' the installer will automatically switch TigerGraph to the latest maintenance version. 
+If any errors occur, the upgrade process will terminate immediately. Follow the instructions in the prompts to manually continue the upgrade or revert to the original version.
+
+. Option `-n`
+
+If the user wants to skip manual input during the upgrade, they can use the `-n` flag, which automatically switches to the new version without requiring any interaction.
+
+[source, gsql]
+----
+$ ./install.sh -Ur -n
+----
+
+.Example:
+[source, gsql]
+----
+./install.sh -Ur -n
+
+   _______                 ______                 __
+  /_  __(_)___ ____  _____/ ____/________ _____  / /_
+   / / / / __ `/ _ \/ ___/ / __/ ___/ __ `/ __ \/ __ \
+  / / / / /_/ /  __/ /  / /_/ / /  / /_/ / /_/ / / / /
+ /_/ /_/\__, /\___/_/   \____/_/   \__,_/ .___/_/ /_/
+       /____/                          /_/
+
+
+Welcome to the TigerGraph platform installer!
+
+[PROGRESS]: 08:10:44 Upgrade TigerGraph platform from existing one ... 
+[NOTE    ]: Obtained version of exist platform: 4.2.0 [OK] 
+[NOTE    ]: Obtained version of new platform: 4.2.1 [OK] 
+----
+
+. Handling Downtime
```

Comment:
I really think numbered list (the "dot" syntax) is not the right choice. We should use section headings instead. This is not a strict sequence of steps. There is a "soft" sequence of 
1. Preparation
2. Running rolling upgrade
3. Reverting if necessary/desired
but I still wouldn't number them.

Within "Running rolling upgrade" there is a mini sequence, because it's a 2-phrase process. We should actually describe this 2-phase process in the intro, so readers understand what is happening and understand their choices for how to run `./install.sh -Ur`:
1. Install the new version on one cluster node at a time but continue to run the old version.
2. Switch all nodes from the old version to the new version.